kmod/src/ioctl.c

```diff
@@ -964,6 +969,10 @@ static long scoutfs_ioc_move_blocks(struct file *file, unsigned long arg)
 	if (mb.len == 0)
 		return 0;
 
+	if (scoutfs_inode_worm_denied(to)) {
```

Comment:
We can't check the inode for enforcement here because the inode fields haven't been refreshed under a cluster lock, they can be old and stale.   The check needs to happen in _data_move_blocks after the _REFRESH lock call.   (And only once :)).